Assets/GameAssets/Boss/BossScripts/Attacks/BaseBossAttack.cs

```diff
@@ -48,6 +53,40 @@ protected virtual void BeginAttack()
         _isAttackActive = true;
     }
 
+    protected virtual void PlayIdleLoop(float second)
+    {
+        if (_idleSFXCoroutine == null)
+        _idleSFXCoroutine = StartCoroutine(LoopIdleCorrectfloat(second));
+    }
+    protected virtual void DestroyIdleLoop()
+    {
+        if (_idleSFXCoroutine != null)
+        {
+            StopCoroutine(_idleSFXCoroutine);
+        }
+    }
+    /// <summary>
+    /// Play heartbeat at controlled rate 
+    /// </summary>
+    /// <returns></returns>
+    private IEnumerator LoopIdleCorrectfloat (float second)
+    {
+        float timer = 0.0f;
+
+        while (true)
+        {
+            if (timer > second)
+            {
+                RuntimeSfxManager.APlayOneShotSfx?.Invoke(FmodSfxEvents.Instance.LimbIdle, transform.position);
+                timer = 0.0f;
+            }
+            timer += Time.deltaTime;
```

Comment:
Could we do just do a WaitForSeconds instead?

Reply:
yes :)